picom: new, 8.2

szclsya opened the pull request:
Topic Description
-----------------
Adding `picom`, an active fork of `compton`, to the repository.

Package(s) Affected
-------------------
N/A

Security Update?
----------------
- [ ] Yes
- [x] No

Architectural Progress
----------------------

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

<!-- TODO: CI to auto-fill architectural progress. -->

Secondary Architectural Progress
--------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

szclsya committed:
uthash: new, 2.1.0

szclsya committed:
picom: new, 8.2

MingcongBai commented on extra-libs/uthash/autobuild/build:
```diff
@@ -0,0 +1,14 @@
+# Taken from https://www.archlinux.org/packages/community/any/uthash/
+
+abinfo "create directory for header files"
+install -dm755 "$PKGDIR"/usr/include/
```
Please enable verbosity this operation.

MingcongBai commented on extra-libs/uthash/autobuild/build:
```diff
@@ -0,0 +1,14 @@
+cd "$SRCDIR/src"
+
+for h in *.h; do
+  install -m 644 "$h" "$PKGDIR"/usr/include/
```
Please enable verbosity this operation.

MingcongBai commented on extra-libs/uthash/autobuild/build:
```diff
@@ -0,0 +1,14 @@
+  install -m 644 "$h" "$PKGDIR"/usr/include/
+done
+
+abinfo "install license file"
```
This step is not necessary and, for future reference, our license and project documentation are all installed in `/usr/share/doc/$PKGNAME`. This is different from Arch's specifications.

MingcongBai commented on extra-libs/uthash/autobuild/build:
```diff
@@ -0,0 +1,14 @@
+# Taken from https://www.archlinux.org/packages/community/any/uthash/
```
Kudos for citation.

MingcongBai commented on extra-libs/uthash/spec:
```diff
@@ -0,0 +1,3 @@
+VER=2.1.0
+SRCTBL="https://github.com/troydhanson/uthash/archive/v$VER.tar.gz"
```
Please switch to our new `spec` format, as specified in our [ACBS Documentation](https://wiki.aosc.io/developer/packaging/acbs/spec-format/). An example format for your reference:

```
VER=2.1.0
SRCS="https://github.com/troydhanson/uthash/archive/v$VER.tar.gz"
CHKSUMS="sha256::152ccd8e64d0f495377232e3964d06c7ec8bb8c3fbd3217f8a5702614f9a669e"
```

MingcongBai commented on extra-x11/picom/spec:
```diff
@@ -0,0 +1,3 @@
+VER=8.2
+SRCTBL="https://github.com/yshui/picom/archive/v8.2.tar.gz"
```
Please switch to our new `spec` format, as specified in our [ACBS Documentation](https://wiki.aosc.io/developer/packaging/acbs/spec-format/).

szclsya committed:
uthash: fix styling issues

- add verbosity
- remove unnecessary license installation
- adapt to new SRC format

szclsya committed:
picom: use new spec format

MingcongBai commented:
@szclsya @Icenowy `picom` fails to launch with a segmentation fault on my PineBook.

```
[ 10/27/20 21:22:30.993 glx_has_extension ERROR ] Failed get GLX extension list.
[ 10/27/20 21:22:31.002 glx_has_extension ERROR ] Failed get GLX extension list.
[ 10/27/20 21:22:31.006 glx_has_extension ERROR ] Failed get GLX extension list.
[ 10/27/20 21:22:31.017 glx_has_extension ERROR ] Failed get GLX extension list.
[ 10/27/20 21:22:31.020 glx_has_extension ERROR ] Failed get GLX extension list.
[ 10/27/20 21:22:31.022 glx_has_extension ERROR ] Failed get GLX extension list.
[ 10/27/20 21:22:31.023 glx_has_extension ERROR ] Failed get GLX extension list.
[ 10/27/20 21:22:31.027 glx_has_extension ERROR ] Failed get GLX extension list.
Segmentation fault (core dumped)
```

Any ideas?

MingcongBai commented:
The issue noted above is addressed with #2431 - a result of a botched Mesa build on AArch64.

Merging.

MingcongBai commented:
Post-Merge Architectural Progress
----------------------

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

<!-- TODO: CI to auto-fill architectural progress. -->

Secondary Architectural Progress
--------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
